### PR TITLE
Test Teardown on Completion of Model Registry Test 1302

### PR DESCRIPTION
**Pull request by tonyxrmdavidson**
This commit will add functionality to remove model-registry and its associated components as the final stage on completion of the testing

### CODE REVIEW

**Comment by bdattoma** [comment]
@tonyxrmdavidson maybe you need to rebase or remove some commits from this PR, there are changes about ROSA HCP which come from different PRs

**Comment by tonyxrmdavidson** [comment]
Successful run in ods-ci jenkins pr-check
https://opendatascience-jenkins-csb-rhods.apps.ocp-c1.prod.psi.redhat.com/job/rhods/job/rhods-ci-pr-test/3254/

**Comment by tonyxrmdavidson** [comment]
Successful PR-Check test run
https://opendatascience-jenkins-csb-rhods.apps.ocp-c1.prod.psi.redhat.com/job/rhods/job/rhods-ci-pr-test/3263/

File: ods_ci/tests/Tests/1300__model_registry/1302_model_registry_model_serving.robot [hunk 1]
```diff
@@ -248,3 +250,27 @@ Run Update Notebook Script
...
     Log    ${result.stderr}
     Should Contain    ${result.stdout}    Modified notebook saved
+
+Remove Model Registry
+    [Documentation]    Run multiple oc delete commands to remove model registry components
+    Run OC Delete Command    oc delete smm default -n ${NAMESPACE_MODEL-REGISTRY}
+    Run OC Delete Command    oc delete -k ${MODELREGISTRY_BASE_FOLDER}/samples/secure-db/mysql-tls
+    Run OC Delete Command    oc delete secret modelregistry-sample-grpc-credential -n ${NAMESPACE_ISTIO}
+    Run OC Delete Command    oc delete secret modelregistry-sample-rest-credential -n ${NAMESPACE_ISTIO}
+    Sleep    30s
```
[the remark this answers is not in the document]
Comment (tonyxrmdavidson) [reply ods_ci/tests/Tests/1300__model_registry/1302_model_registry_model_serving.robot]: I removed the sleep as it was not needed.

**Review by nehachopra27 — APPROVED** [review]
LGTM

**Review by manosnoam — CHANGES_REQUESTED** [review]

File: ods_ci/tests/Tests/1300__model_registry/1302_model_registry_model_serving.robot [hunk 2]
```diff
@@ -248,3 +250,25 @@ Run Update Notebook Script
...
+
+Remove Model Registry
+    [Documentation]    Run multiple oc delete commands to remove model registry components
+    Run OC Delete Command    oc delete smm default -n ${NAMESPACE_MODEL-REGISTRY}
+    Run OC Delete Command    oc delete -k ${MODELREGISTRY_BASE_FOLDER}/samples/secure-db/mysql-tls
+    Run OC Delete Command    oc delete secret modelregistry-sample-grpc-credential -n ${NAMESPACE_ISTIO}
+    Run OC Delete Command    oc delete secret modelregistry-sample-rest-credential -n ${NAMESPACE_ISTIO}
+    Run OC Delete Command    oc delete namespace ${NAMESPACE_MODEL-REGISTRY} --force
+
+Run OC Delete Command
```
Comment (manosnoam) [inline ods_ci/tests/Tests/1300__model_registry/1302_model_registry_model_serving.robot:262]: Can you please rename this Keyword to something like `Run And Verify Command` and place it in the `ods_ci/tests/Resources/Common.robot` file, above the similar KW `Run And Watch Command` (line 435) ?

**Review by manosnoam — CHANGES_REQUESTED** [review]

**Comment by tonyxrmdavidson** [comment]
Screenshot 2024-08-15 at 16 21 02

File: ods_ci/tests/Tests/1300__model_registry/1302_model_registry_model_serving.robot (hunk 2 above)
Comment (tonyxrmdavidson) [reply ods_ci/tests/Tests/1300__model_registry/1302_model_registry_model_serving.robot]: Screenshot 2024-08-15 at 16 21 02

Renamed keyword and moved it to Common.robot

**Review by manosnoam — APPROVED** [review]

**Review by nehachopra27 — APPROVED** [review]

**Pull request merged**